### DISCUSSION TITLE
Do we have any option to read and validate RDS proxy and its endpoint using terratest golang

### DISCUSSION THREAD

**Discussion by pandu-bhojaraj-mf** [gruntwork-io/knowledge-base#346] [Help]
Do we have any option to read and validate RDS proxy and its endpoint using terratest golang

Hi Team, 

I have got RDS proxy and endpoint got deployed into my AWS setup and i got a task to validate those , i am able to validate them manually but looking for automated option if anything available. **I am looking for supported functions to read RDS proxy and endpoints using terratest golang** , please help me on this part.

<img width="940" alt="image" src="https://user-images.githubusercontent.com/73541850/162368073-9da7f4f9-db12-4aa1-b0fe-6aea8479bf83.png">

**Comment by denis256** [comment]
Hi,
from a search in the code base, looks like such functions aren't available now

**Comment by rhoboat** [comment]
Hi @pandu-bhojaraj-mf!

I understand you want to validate that you can access the RDS endpoints, during your terratest run, is that right?

Terratest validation commands
----------------------------------
- Can you use something from this set of functions, to run a [custom command](https://github.com/gruntwork-io/terratest/blob/c00bbe0aaae34683a81e52d2515388758e7e431c/modules/shell/command.go#L48)?
- Or can you use this [http-helper](https://github.com/gruntwork-io/terratest/blob/c00bbe0aaae34683a81e52d2515388758e7e431c/modules/http-helper/http_helper.go#L62) to check the endpoint?
- To read the endpoint, can you use the [terraform output methods](https://github.com/gruntwork-io/terratest/blob/master/modules/terraform/output.go), if your module exposes it as an output?

Test structure
-----------------
In your test, you'll also need to spin up (apply) the infrastructure, run the validation, and then destroy. For the overall structure of the test, please look at how we do the Test Stages like in this example: https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/bf4330b49ebd9b8de18b6a622238ea248c805830/test/mgmt/tailscale_subnet_router_test.go#L169

This example uses http_helper for the validation.

Access to the RDS endpoint
----------------------------
You'll also need to make sure you can access the RDS endpoint from the test. RDS is probably running in a private subnet of your VPC, which means the test instance needs to make a connection to the VPC. Just make sure you have thought about this aspect, as well.

**Reply by pandu-bhojaraj-mf** [reply]
I will have a look into this Thanks
